alchemist-graphql/src/jvmMain/kotlin/it/unibo/alchemist/boundary/launchers/GraphQLServerLauncher.kt

```diff
@@ -31,16 +32,17 @@ class GraphQLServerLauncher @JvmOverloads constructor(
 ) : SimulationLauncher() {
     override fun launch(loader: Loader) {
         val simulation: Simulation<Any, Nothing> = prepareSimulation(loader, emptyMap<String, Any>())
+        simulation.addOutputMonitor(EnvironmentSubscriptionMonitor())
         startServer(simulation)
     }
 
-    private fun <T, P : Position<P>> startServer(
+    private fun <T, P : Position<out P>> startServer(
         simulation: Simulation<T, P>,
-        serverDispatcher: CoroutineDispatcher = Dispatchers.IO,
+        serverDispatcher: CoroutineDispatcher = Dispatchers.Default,
```

Comment:
add a secondary constructor taking a string and selecting the Dispatcher based on the string

Reply:
This is not easy as it seems, due to detekt complaining about using coroutines dispatchers without dependency injection. I think that @AngeloFilaseta has encountered this issue too.

Reply:
What is the detekt rule?

Reply:
It is `detektJvmMain`, it fails with the message: "Dispatcher Default/IO/Unconfined is used without dependency injection. [InjectDispatcher]"

Reply:
Sorry I thought it was a constructor. My bad